tests/testthat/setup-workflow-ref.R

```diff
@@ -194,7 +194,16 @@ create_rlg_models <- function() {
   )
 }
 
-cleanup <- function() {
+clean_test_enviroment <- function(.f, env = parent.frame())
+{
+  cleanup(env)
+  .f()
+  withr::defer(cleanup(env), envir = env)
+}
```

Comment:
Nice, this is looking as I expected.  I guess this approach looked okay when you played around with it?

Reply:
Debugged on call.  Tentative answer is that `cleanup` wasn't doing enough cleanup, unrelated to environment handling.